src/pe/dynreloc.rs

```diff
@@ -0,0 +1,1258 @@
+    }
+}
+
+impl IndirectControlTransferDynReloc {
+    /// Creates a new DynRelocIndirControlTransfer from individual field values
+    pub fn new(
+        page_relative_offset: u16,
+        indirect_call: bool,
+        rex_w_prefix: bool,
+        cfg_check: bool,
+    ) -> Self {
+        let page_relative_offset = page_relative_offset & 0xFFF; // 12 bits
+        let indirect_call = if indirect_call { 1u16 } else { 0u16 };
+        let rex_w_prefix = if rex_w_prefix { 1u16 } else { 0u16 };
+        let cfg_check = if cfg_check { 1u16 } else { 0u16 };
+        // Reserved bit is always 0
+
+        let value =
+            page_relative_offset | (indirect_call << 12) | (rex_w_prefix << 13) | (cfg_check << 14);
+        // bit 15 (Reserved) is left as 0
+
+        Self(value)
+    }
+
+    /// Creates from raw u16 value
+    pub fn from_raw(value: u16) -> Self {
+        Self(value)
+    }
+
+    /// Gets the PageRelativeOffset field (bits 0-11)
+    pub fn page_relative_offset(&self) -> u16 {
+        self.0 & 0xFFF
+    }
+
+    /// Sets the PageRelativeOffset field (bits 0-11)
+    pub fn set_page_relative_offset(&mut self, offset: u16) {
+        let offset = offset & 0xFFF;
+        self.0 = (self.0 & !0xFFF) | offset;
+    }
+
+    /// Gets the IndirectCall field (bit 12)
+    /// True if this is an indirect call, false if direct
+    pub fn indirect_call(&self) -> bool {
+        (self.0 >> 12) & 1 != 0
+    }
+
+    /// Sets the IndirectCall field (bit 12)
+    pub fn set_indirect_call(&mut self, indirect: bool) {
+        if indirect {
+            self.0 |= 1 << 12;
+        } else {
+            self.0 &= !(1 << 12);
+        }
+    }
+
+    /// Returns true if this is an indirect call
+    pub fn is_indirect_call(&self) -> bool {
+        self.indirect_call()
+    }
+
+    /// Returns true if this is a direct call
+    pub fn is_direct_call(&self) -> bool {
+        !self.indirect_call()
+    }
+
+    /// Gets the RexWPrefix field (bit 13)
+    /// True if the instruction uses REX.W prefix (64-bit operand)
+    pub fn rex_w_prefix(&self) -> bool {
+        (self.0 >> 13) & 1 != 0
+    }
+
+    /// Sets the RexWPrefix field (bit 13)
+    pub fn set_rex_w_prefix(&mut self, rex_w: bool) {
+        if rex_w {
+            self.0 |= 1 << 13;
+        } else {
+            self.0 &= !(1 << 13);
+        }
+    }
+
+    /// Returns true if the instruction has 64-bit operand size (REX.W prefix)
+    pub fn is_64bit_operand(&self) -> bool {
+        self.rex_w_prefix()
+    }
+
+    /// Returns true if the instruction has 32-bit operand size (no REX.W prefix)
+    pub fn is_32bit_operand(&self) -> bool {
+        !self.rex_w_prefix()
+    }
+
+    /// Gets the CfgCheck field (bit 14)
+    /// True if Control Flow Guard check is required
+    pub fn cfg_check(&self) -> bool {
+        (self.0 >> 14) & 1 != 0
+    }
+
+    /// Sets the CfgCheck field (bit 14)
+    pub fn set_cfg_check(&mut self, cfg_check: bool) {
```

Comment:
I might opt for the builder pattern here, but I don't know how usage patterns are generally, do you normally mutate it a few times occasionally in different situations or is it generally constructed with these flags once? If latter I'd suggest to use builder pattern since it's more convenient. If it's mutated in different situations/conditions, set style is probably better